modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/ApplicationInterfaceRepository.java

```diff
@@ -0,0 +1,256 @@
+        return applicationModule;
+    }
+
+    @Override
+    public ApplicationInterfaceDescription getApplicationInterface(String interfaceId) throws AppCatalogException {
+        Map<String, Object> queryParameters = new HashMap<>();
+        queryParameters.put(DBConstants.ApplicationInterface.APPLICATION_INTERFACE_ID, interfaceId);
+        ApplicationInterfaceDescription applicationInterfaceDescription = select(QueryConstants.FIND_APPLICATION_INTERFACE, -1, 0, queryParameters).get(0);
+        return applicationInterfaceDescription;
+    }
+
+    @Override
+    public List<ApplicationModule> getApplicationModules(Map<String, String> filters) throws AppCatalogException {
+        ApplicationModuleRepository applicationModuleRepository = new ApplicationModuleRepository();
+        if(filters.containsKey(DBConstants.ApplicationModule.APPLICATION_MODULE_NAME)) {
+            Map<String, Object> queryParameters = new HashMap<>();
+            queryParameters.put(DBConstants.ApplicationModule.APPLICATION_MODULE_NAME, filters.get(DBConstants.ApplicationModule.APPLICATION_MODULE_NAME));
+            List<ApplicationModule> applicationModuleList = applicationModuleRepository.select(QueryConstants.FIND_APPLICATION_MODULES_FOR_APPLICATION_MODULE_NAME, -1,0, queryParameters);
+            return applicationModuleList;
+        } else {
+            logger.error("Unsupported field name for app module.", new IllegalArgumentException());
+            throw new IllegalArgumentException("Unsupported field name for app module.");
+        }
+    }
+
+    @Override
+    public List<ApplicationModule> getAllApplicationModules(String gatewayId) throws AppCatalogException {
+        ApplicationModuleRepository applicationModuleRepository = new ApplicationModuleRepository();
+        Map<String, Object> queryParameters = new HashMap<>();
+        queryParameters.put(DBConstants.ApplicationModule.GATEWAY_ID, gatewayId);
+        List<ApplicationModule> applicationModuleList = applicationModuleRepository.select(QueryConstants.FIND_APPLICATION_MODULES_FOR_GATEWAY_ID, -1, 0, queryParameters);
+        return applicationModuleList;
+    }
+
+    @Override
+    public List<ApplicationInterfaceDescription> getApplicationInterfaces(Map<String, String> filters) throws AppCatalogException {
+        if(filters.containsKey(DBConstants.ApplicationInterface.APPLICATION_NAME)) {
+            Map<String, Object> queryParameters = new HashMap<>();
+            queryParameters.put(DBConstants.ApplicationInterface.APPLICATION_NAME, filters.get(DBConstants.ApplicationInterface.APPLICATION_NAME));
+            List<ApplicationInterfaceDescription> applicationInterfaceDescriptionList = select(QueryConstants.FIND_APPLICATION_INTERFACES_FOR_APPLICATION_NAME, -1,0, queryParameters);
+            return applicationInterfaceDescriptionList;
+        } else {
+            logger.error("Unsupported field name for app interface.", new IllegalArgumentException());
```

Comment:
You do not have to push a new IllegalArgumentException to error logs. Just print the log line.